sql_generators/glean_usage/templates/event_flow_monitoring_aggregates_v1.script.sql

```diff
@@ -0,0 +1,187 @@
+            TIMESTAMP_ADD(
+              submission_timestamp,
+          -- limit event.timestamp, otherwise this will cause an overflow
+              INTERVAL LEAST(event_timestamp, 20000000000000) MILLISECOND
+            ) AS timestamp,
+            "{{ app }}" AS normalized_app_name,
+            client_info.app_channel AS channel
+          FROM
+            `moz-fx-data-shared-prod.{{ app }}.events_unnested`,
+            UNNEST(event_extra) AS ext
+          WHERE
+            DATE(submission_timestamp) = @submission_date
+            AND ext.key = "flow_id"
+        {% elif dataset_id in ["accounts_frontend", "accounts_backend"] %}
+          SELECT DISTINCT
+            @submission_date AS submission_date,
+            metrics.string.session_flow_id AS flow_id,
+            NULL AS category,
+            metrics.string.event_name AS name,
+            submission_timestamp AS timestamp,
+            "{{ app }}" AS normalized_app_name,
+            client_info.app_channel AS channel
+          FROM
+            `moz-fx-data-shared-prod.{{ app }}.accounts_events`
+          WHERE
+            DATE(submission_timestamp) = @submission_date
+            AND metrics.string.session_flow_id != ""
+        {% endif %}
+      {% endfor %}
+    ),
+    -- determine events that belong to the same flow
+    new_event_flows AS (
+      SELECT
+        @submission_date AS submission_date,
+        flow_id,
+        normalized_app_name,
+        channel,
+        ARRAY_AGG(
+          STRUCT(category AS category, name AS name, timestamp AS timestamp)
+          ORDER BY
+            timestamp
+        ) AS events
+      FROM
+        all_app_events
+      GROUP BY
+        flow_id,
+        normalized_app_name,
+        channel
+    ),
+    unnested_events AS (
+      SELECT
+        new_event_flows.*,
+        event,
+        event_offset
+      FROM
+        new_event_flows,
+        UNNEST(events) AS event
+        WITH OFFSET AS event_offset
+    ),
+    -- create source -> target event pairs based on the order of when the events were seen
+    source_target_events AS (
+      SELECT
+        prev_event.flow_id,
+        prev_event.normalized_app_name,
+        prev_event.channel,
+        ARRAY_AGG(
+          STRUCT(prev_event.event AS source, cur_event.event AS target)
+          ORDER BY
+            prev_event.event.timestamp
+        ) AS events
+      FROM
+        unnested_events AS prev_event
+      INNER JOIN
+        unnested_events AS cur_event
+      ON
+        prev_event.flow_id = cur_event.flow_id
+        AND prev_event.event_offset = cur_event.event_offset - 1
+      GROUP BY
+        flow_id,
+        normalized_app_name,
+        channel
+    )
+    SELECT
+      @submission_date AS submission_date,
+      flow_id,
+      normalized_app_name,
+      channel,
+      ARRAY_AGG(event ORDER BY event.source.timestamp) AS events,
+      -- create a flow hash that concats all the events that are part of the flow
```

Comment:
This might break for flows long enough to exceed some BQ limit. Similar thing was happening in some of the tables generated with `events_daily` sql generator.

Reply:
Good point. I guess a workaround is to limit flows to having a maximum number of events here